training/feature_extractor.py

```
@@ -0,0 +1,351 @@
+            np.save(feature_path, feature_stack.astype(np.float32))
+
+            # Cache result
+            self.cache[cache_key] = feature_stack
+
+            # Save cache periodically
+            if len(self.cache) % 100 == 0:
+                self._save_persistent_cache()
+
+            return feature_stack
+
+        except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)